libparsec/crates/platform_pki/src/windows/mod.rs

```diff
@@ -27,36 +27,30 @@ fn open_store() -> std::io::Result<CertStore> {
     CertStore::open_current_user("My")
 }
 
-impl CertificateHash {
-    fn get_hash_algo(&self) -> HashAlgorithm {
-        match self {
-            CertificateHash::SHA256(..) => HashAlgorithm::sha256(),
-        }
+fn get_hash_algo(&hash: CertificateHash) -> HashAlgorithm {
+    match hash {
+        CertificateHash::SHA256(..) => HashAlgorithm::sha256(),
     }
 }
 
-impl PartialEq<CertContext> for CertificateHash {
-    fn eq(&self, other: &CertContext) -> bool {
-        let Ok(cert_hash) = other.fingerprint(self.get_hash_algo()) else {
-            return false;
-        };
-        match self {
-            CertificateHash::SHA256(expected) => expected.as_ref() == cert_hash.as_slice(),
-        }
+fn eq(&hash: CertificateHash, other: &CertContext) -> bool {
```

Comment:
Any suggestions for a better name ?

(it had to become just an function because neither CertificateHash nor Certcontext are defined in this crate)